misk/src/main/kotlin/misk/client/ClientMetricsInterceptor.kt

```diff
@@ -13,14 +16,20 @@ class ClientMetricsInterceptor internal constructor(
 ) : ClientNetworkInterceptor {
 
   override fun intercept(chain: ClientNetworkChain): Response {
-    val (elapsedTime, result) = timed { chain.proceed(chain.request) }
-    val elapsedMillis = elapsedTime.toMillis().toDouble()
-
-    requestDuration.record(elapsedMillis, actionName, "all")
-    requestDuration.record(elapsedMillis, actionName, "${result.code / 100}xx")
-    requestDuration.record(elapsedMillis, actionName, "${result.code}")
-
-    return result
+    val stopwatch = Stopwatch.createStarted(Ticker.systemTicker())
+    try {
+      val result = chain.proceed(chain.request)
+      val elapsedMillis = stopwatch.stop().elapsed(TimeUnit.MILLISECONDS).toDouble()
+      requestDuration.record(elapsedMillis, actionName, "all")
+      requestDuration.record(elapsedMillis, actionName, "${result.code / 100}xx")
+      requestDuration.record(elapsedMillis, actionName, "${result.code}")
+      return result
+    } catch (e: SocketTimeoutException) {
+      val elapsedMillis = stopwatch.stop().elapsed(TimeUnit.MILLISECONDS).toDouble()
+      requestDuration.record(elapsedMillis, actionName, "all")
+      requestDuration.record(elapsedMillis, actionName, "timeout")
```

Comment:
is this that useful? It's essentially what the client timeout is set to, no?

Reply:
Yeah, the histogram metric is not very interesting other than validate you're timing out on the expected time. I was more interested to know how many times outs the client is seeing. Also, it consistent with the other metrics. Let me know if you want to use other metric.

Reply:
I think it’s useful in that if it spikes it tells us something